backend/rules/attackpath/PubliclyExposedVmAdminCVECritical.go

```diff
@@ -0,0 +1,185 @@
+package attackpath
+
+import (
+	"fmt"
+
+	"github.com/Zeus-Labs/ZeusCloud/rules/types"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+)
+
+type PubliclyExposedVmAdminCVECritical struct{}
+
+func (PubliclyExposedVmAdminCVECritical) UID() string {
+	return "attackpath/publicly_exposed_vm_admin_permissions_crtitical_cve"
+}
+
+func (PubliclyExposedVmAdminCVECritical) Description() string {
+	return "Publicly exposed VM instance with effective admin permissions and critical cve."
+}
+
+func (PubliclyExposedVmAdminCVECritical) Severity() types.Severity {
+	return types.Critical
+}
+
+func (PubliclyExposedVmAdminCVECritical) RiskCategories() types.RiskCategoryList {
+	return []types.RiskCategory{
+		types.PubliclyExposed,
+		types.IamMisconfiguration,
+		types.Vulnerability,
+	}
+}
+
+// EC2 is considered publicly exposed if
+// - it's directly exposed through security group / IP
+// - it's exposed through an ELBv2
+// TODO: Add other mechanisms of exposure like ELBv1?
+func (PubliclyExposedVmAdminCVECritical) Execute(tx neo4j.Transaction) ([]types.Result, error) {
+	records, err := tx.Run(
+		`MATCH (a:AWSAccount{inscope: true})-[:RESOURCE]->(e:EC2Instance)
+		OPTIONAL MATCH
+			(:IpRange{id: '0.0.0.0/0'})-[:MEMBER_OF_IP_RULE]->
+			(:IpPermissionInbound)-[:MEMBER_OF_EC2_SECURITY_GROUP]->
+			(instance_group:EC2SecurityGroup)<-[:MEMBER_OF_EC2_SECURITY_GROUP|NETWORK_INTERFACE*..2]-(e)-[:HAS_VULNERABILITY]->(ec2CVE:CVE)
```

Comment:
How do we know the CVE is critical? Is there a field in the cve node with a cvss score to figure out severity? Need to have a check that the CVE is critical.